pkg/cosign/bundle/bundle.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"github.com/sigstore/rekor/pkg/generated/models"
+)
+
+type Bundle struct {
+	VerificationData
+	VerificationMaterial
+}
+
+// VerificationData contains extra data that can be used to verify things
+// such as transparency logs and timestamped verifications.
+type VerificationData struct {
+	// RekorPayload holds metadata about recording a Signature's ephemeral key to
+	// a Rekor transparency log.
+	// Use Payload instead of TransparencyLogEntry to keep backwards compatibility.
+	Payload RekorPayload
+
+	// TimestampVerificationData holds metadata about a timestamped verification.
+	TimestampVerificationData
+}
+
+// VerificationMaterial captures details on the materials used to verify
+// signatures or any additional timestamped verifications.
+type VerificationMaterial struct {
+	// A chain of X.509 certificates.
+	CertBytes []byte
+
+	// PublicKeyIdentifier optional unauthenticated hint on which key to use.
+	PublicKeyIdentifier string
+}
+
+// TimestampVerificationData contains various timestamped data following RFC3161.
+type TimestampVerificationData struct {
+	// SignedEntryTimestamp holds metadata about a timestamped counter signature over the artifacts signature.
+	SignedEntryTimestamp []byte
+
+	// EntryTimestampAuthority contains the recorded entry from timestamp authority server.
+	EntryTimestampAuthority []byte
+}
+
+func EntryToBundle(tLogEntry *models.LogEntryAnon, signedEntryTimestamp, entryTimestampAuthority, certBytes []byte, pubKeyID string) *Bundle {
+	b := &Bundle{}
+	// If none of the verification data is configured then return nil
+	if (tLogEntry == nil || tLogEntry.Verification == nil) && len(entryTimestampAuthority) == 0 {
+		return nil
```

Comment:
why return nil over returning an empty bundle? id be worried about accidental dereferences later

Reply:
I agree with you but `tLogEntry == nil || tLogEntry.Verification == nil` is reusing the old behavior that also returned a nil bundle.